pkg/resourcemapping/operator_resource_mapping.go

```diff
@@ -0,0 +1,340 @@
+		Resource: ormResource,
+	}
+	ormCRs, err := ormClient.dynClient.Resource(groupVersionRes).List(metav1.ListOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to get OperatorResourceMapping CRs: %v", err)
+	}
+	return ormCRs.Items, nil
+}
+
+func (ormClient *ORMClient) getOperatorCRsFromCRD(crdName, namespace string) ([]unstructured.Unstructured, schema.GroupVersionResource, error) {
+	crds := ormClient.apiExtClient.CustomResourceDefinitions()
+	crd, err := crds.Get(crdName, metav1.GetOptions{})
+	if err != nil {
+		return nil, schema.GroupVersionResource{}, err
+	}
+	groupVersionRes := schema.GroupVersionResource{
+		Group:    crd.Spec.Group,
+		Version:  crd.Spec.Versions[0].Name,
+		Resource: crd.Spec.Names.Plural,
+	}
+	crs, err := ormClient.dynClient.Resource(groupVersionRes).Namespace(namespace).List(metav1.ListOptions{})
+	if err != nil {
+		return nil, groupVersionRes, err
+	}
+	return crs.Items, groupVersionRes, nil
+}
+
+func (ormClient *ORMClient) populateORMTemplateMap(ormCR unstructured.Unstructured) (map[string]ORMTemplate, error) {
+	ormCRName := ormCR.GetName()
+	ormTemplateMap := make(map[string]ORMTemplate)
+	resourceMappings, _, err := unstructured.NestedSlice(ormCR.Object, ormResourceMappingsPath...)
+	if err != nil {
+		return nil, fmt.Errorf("no data under '%s' in ORM CR %s: %v", util.JSONPath(ormResourceMappingsPath), ormCRName, err)
+	}
+	for _, resourceMapping := range resourceMappings {
+		rm, ok := resourceMapping.(map[string]interface{})
+		if !ok {
+			glog.Errorf("ResourceMapping does not have the expected 'map[string]interface{}' structure in ORM CR %s", ormCRName)
+			continue
+		}
+		srcResourceKind, found, err := unstructured.NestedString(rm, srcResourceSpecKindPath...)
+		if err != nil || !found {
+			glog.Errorf("Value is not found under '%s' under '%s' in ORM CR %s", util.JSONPath(srcResourceSpecKindPath),
+				util.JSONPath(ormResourceMappingsPath), ormCRName)
+			continue
+		}
+		componentNames, found, err := unstructured.NestedStringSlice(rm, srcResourceSpecComponentNamesPath...)
+		if err != nil || !found {
+			glog.Errorf("Value is not found under '%s' under '%s' in ORM CR %s", util.JSONPath(srcResourceSpecComponentNamesPath),
+				util.JSONPath(ormResourceMappingsPath), ormCRName)
+			continue
+		}
+		resourceMappingTemplates, found, err := unstructured.NestedSlice(rm, resourceMappingTemplatesPath...)
+		if err != nil || !found {
+			glog.Errorf("Value not found under '%s' under '%s' in ORM CR %s", util.JSONPath(resourceMappingTemplatesPath),
+				util.JSONPath(ormResourceMappingsPath), ormCRName)
+			continue
+		}
+
+		rmTemplates := make([]map[string]interface{}, 0, len(resourceMappingTemplates))
+		for _, resourceMappingTemplate := range resourceMappingTemplates {
+			rmTemplate, ok := resourceMappingTemplate.(map[string]interface{})
+			if !ok {
+				glog.Errorf("resourceMappingTemplate does not have the expected 'map[string]interface{}' structure in ORM CR %s", ormCRName)
```

Comment:
Same as above, use the index. In this case it would be 2 indexes printed.

Reply:
Thanks. Will use the index in the error message